src/odemis/driver/lakeshore.py

```diff
@@ -163,45 +220,72 @@ def _openSerialPort(port, baudrate=57600):
 
         return ser
 
-    def _findDevice(self, ports):
+    def _checkDeviceCompatibility(self):
+        """
+        Checks the device manifacturer and model
+        raises:
+         IOError: Not supported device model or manifacturer
+         There might be other errors such as timeout error
+        """
+        manufacturer, md, _, _ = self.GetIdentifier()  # if value is incorrect, will throw an exception wile unpacking
+
+        if manufacturer != "LSCI":
+            raise IOError("Invalid device manufacturer")
+
+        compatible_devices = ["MODEL335", "MODEL350"]
+        if md not in compatible_devices:
+            raise IOError("Incompatible device model detected. "
+                          "Compatible models are %s." % (compatible_devices,))
+
+    def _findDevice(self, bus_pattern):
         """
         Look for a compatible device
-        ports (str): pattern for the port name
+        bus_pattern (str): pattern for the port name
         baudrate (0<int)
         return:
            (str): the name of the port used
            Note: will also update ._file and ._serial
         raises:
             IOError: if no devices are found
         """
-        # For debugging purpose
-        if ports == "/dev/fake":
-            self._serial = LakeshoreSimulator(timeout=1)
-            self._file = None
-            return ports
 
-        if os.name == "nt":
-            raise NotImplementedError("Windows not supported")
+        # Connection via ethernet cable
+        if not bus_pattern.startswith("/"):
+            self._accesser = IPBusAccesser(bus_pattern)
+
+            manufacturer, md, _, _ = self.GetIdentifier()  # if value is incorrect, will throw an exception wile unpacking
+
+            if manufacturer != "LSCI":
+                raise IOError("Invalid device manufacturer")
+
+            compatible_devices = ["MODEL335", "MODEL350"]
+            if md not in compatible_devices:
+                raise IOError("Incompatible device model detected. "
+                              "Compatible models are %s." % (compatible_devices,))
+
+            return
```

Comment:
It's important that if the connection fails (eg, because the device is turned off, or not connected to the network), the function raises HwError(). See how it's done for the serial port case. This allows the backend to know that the device is not ready. It'll display the error to the user, who has the possibility to fix the issue, and automatically the backend will retry.